src/datatypes/quantity.js

```diff
@@ -107,18 +107,30 @@ class Quantity {
 
   multiplyDivide(other, operator) {
     if (other != null && other.isQuantity) {
-      const a = this.unit != null ? this : new Quantity(this.value, '1');
-      const b = other.unit != null ? other : new Quantity(other.value, '1');
-      const can_val = a.to_ucum();
-      const other_can_value = b.to_ucum();
-      const ucum_value = ucum_multiply(can_val, [[operator, other_can_value]]);
-      if (overflowsOrUnderflows(ucum_value.value)) {
-        return null;
-      }
-      try {
-        return new Quantity(ucum_value.value, units_to_string(ucum_value.units));
-      } catch (e) {
-        return null;
+      if (other.unit === '1' || other.unit === '') {
+        const value = operator === '/' ? this.value / other.value : this.value * other.value;
+        if (overflowsOrUnderflows(value)) {
+          return null;
+        }
+        try {
+          return new Quantity(decimalAdjust('round', value, -8), coalesceToOne(this.unit));
+        } catch (e) {
+          return null;
+        }
+      } else {
+        const a = this.unit != null ? this : new Quantity(this.value, '1');
+        const b = other.unit != null ? other : new Quantity(other.value, '1');
```

Comment:
I don't know if this can happen, but if `other.unit` is null, it will convert to `'1'`.  If that happens will we end up w/ a similar issue as before.

Or, in other words, should line 110 above also check for `other.unit == null`?

Reply:
Done!